OpenRA.Mods.Common/UtilityCommands/UpgradeMapCommand.cs

```diff
@@ -23,13 +25,25 @@ public bool ValidateArguments(string[] args)
 			return args.Length >= 3;
 		}
 
-		[Desc("MAP", "CURRENTENGINE", "Upgrade map rules to the latest engine version.")]
-		public void Run(ModData modData, string[] args)
+		delegate void UpgradeAction(int engineVersion, ref List<MiniYamlNode> nodes, MiniYamlNode parent, int depth);
+
+		static void ProcessYaml(Map map, IEnumerable<string> files, int engineDate, UpgradeAction processFile)
```

Comment:
This will be introduced and used by #10839.  For now it's just a rebase failure.